### DISCUSSION TITLE
How to communicate with platform (Android and iOS)?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#1302] [Q&A]
Hi,

How to communicate with platform (Android and iOS)?

For example: I want call Facebook, Google, Firebase and other frameworks functions.

How is the best current way to do this?

Thanks.

**Comment by rh101** [comment]
> For example: I want call Facebook, Google, Firebase and other frameworks functions.

A few have their own SDKs/libraries that you can link to your application, such as the Firebase C++ iOS and Android libaries [here](https://firebase.google.com/docs/cpp/setup?platform=android).  Facebook has SDKs for iOS (ObjC/Swift) and Android (Java), so you would need to write your own wrappers around those if you want to call them from C++.  For other services (Google etc), or ones that don't have suitable SDKs to use, and assuming they have web APIs available that you can call over HTTPs, then it's as easy as using the `HttpClient` in Axmol to create the web requests.

If you want an example of how wrappers are written for Java/ObjC and similar for the above framework SDKs, then have a look at [this](https://github.com/newnon/avalon) project.
